addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -241,14 +241,17 @@ def action_done(self):
                     picking.batch_id.id,
                     picking.batch_id.name))
 
+        # Run sanity_check as a batch and ignore the one in button_validate() since it is done here.
+        pickings._sanity_check(separate_pickings=False)
+        # Skip sanity_check in pickings button_validate() & remove 'waiting' pickings from the batch
+        context = {'skip_sanity_check': True, 'pickings_to_detach': waiting_pickings.ids}
```

Comment:
This is maybe a use case we can ignore, but if you do the following:
- force a Done amount on a "waiting" picking (within the batch) 
- validate the batch
- don't create a backorder
Then done amount on the waiting picking won't be validated and the picking will be removed from the batch. This behavior could maybe be confusing to users who do this?

Reply:
Indeed. By removing all `waiting` pickings right at the start, it would exclude them from the process, even though they had some quantity done.
Changed it so it only removes the *empty* `waiting` pickings instead.